src/core/geometry/qgslinestring.cpp

```diff
@@ -502,6 +502,93 @@ bool QgsLineString::boundingBoxIntersects( const QgsRectangle &rectangle ) const
   return QgsCurve::boundingBoxIntersects( rectangle );
 }
 
+bool QgsLineString::boundingBoxIntersects( const QgsBox3D &box3d ) const
+{
+  if ( mX.empty() )
+    return false;
+
+  if ( mZ.empty() )
+    return boundingBoxIntersects( QgsRectangle( box3d.xMinimum(), box3d.yMinimum(), box3d.xMaximum(), box3d.yMaximum() ) );
+
+  if ( !mBoundingBox.isNull() )
+  {
+    return mBoundingBox.intersects( box3d );
+  }
+  const int nb = mX.size();
+
```

Comment:
Given this is near duplicate code to `QgsLineString::boundingBoxIntersects( const QgsRectangle &rectangle )`, can you add some comments to both that "if one of these functions is changed then remember to also update the other accordingly".

Reply:
ok